Source/Core/VideoBackends/OGL/Render.cpp

```diff
@@ -1269,44 +1275,54 @@ void Renderer::SetBlendingState(const BlendingState& state)
   bool useDualSource =
       state.usedualsrc && g_ActiveConfig.backend_info.bSupportsDualSourceBlend &&
       (!DriverDetails::HasBug(DriverDetails::BUG_BROKEN_DUAL_SOURCE_BLENDING) || state.dstalpha);
+  // Only use shader blend if we need to and we don't support dual-source blending directly
+  bool useShaderBlend = !useDualSource && state.usedualsrc && state.dstalpha &&
```

Comment:
Small nitpick: I'd prefer if this will be checked in BlendingState::Generate, but I see the issue that this won't fit to the way how you're fetching this data again in the PixelShader UID generation. Maybe add an optional flag for querying for the shader. But that isn't nice either...
So feel free to ignore me here...

Reply:
If we're keeping the idea of backend-agnostic state UIDs, it could be an issue. I suppose we could still use "shader blending" bits, and then mask them out in the backend.